packages/headless/src/features/folding/folding-actions.ts

```diff
@@ -73,7 +66,8 @@ export const registerFolding = createAction(
 
 export type StateNeededByLoadCollection = ConfigurationSection &
   FoldingSection &
-  QuerySection;
+  QuerySection &
+  InsightConfigurationSection;
```

Comment:
Am I correct that this `loadCollection` action should only be used by the search use case, and the insight use case uses a different one?

If so, you should revert this change.
```suggestion
  QuerySection;
```

Reply:
loadCollection is used in both useCases, but the insight one has the InsightConfigurationSection on top of the what the folding-action. I thought I could declare it in one and import it, but following your other comment in insight-folding-actions.ts it would be better maybe to declare it in each file